break not working in a for loop nested under a condition

This program fails
```
actor Thing():
    def foo():
        return 1

actor main(env):
    var thingies = [Thing()]

    for thing in thingies:
        bar = thing.foo()
        if 1 > 0:
            break

    env.exit(0)
```

like this:
```
kll@ThinkYoga:~/terastream/acton/test$ ../actonc --root main for-break.act
/home/kll/terastream/acton/modules/test/for-break.c: In function ‘test$_$for_45_break$$l$5c$7cont’:
/home/kll/terastream/acton/modules/test/for-break.c:126:9: error: break statement not within loop or switch
  126 |         break;
      |         ^~~~~
/usr/bin/ld: /tmp/ccyWB2dw.o: in function `$ROOTINIT':
/home/kll/terastream/acton/modules/test/for-break.root.c:3: undefined reference to `test$_$for_45_break$$__init__'
/usr/bin/ld: /tmp/ccyWB2dw.o: in function `$ROOT':
/home/kll/terastream/acton/modules/test/for-break.root.c:6: undefined reference to `test$_$for_45_break$$main$new'
collect2: error: ld returned 1 exit status
kll@ThinkYoga:~/terastream/acton/test$ 
```

I think something special happens when an actor is involved, like `thing` in our code. If we remove that or replace it with a simple variable assignment, everything works just fine:
```
actor Thing():
    def foo():
        return 1

actor main(env):
    var thingies = [Thing()]

    for thing in thingies:
        bar = "hello"
        if 1 > 0:
            break

    env.exit(0)
```
```
kll@ThinkYoga:~/terastream/acton/test$ ../actonc --root main for-break.act
kll@ThinkYoga:~/terastream/acton/test$ echo $?
0
kll@ThinkYoga:~/terastream/acton/test$
```

Similarly moving the break out of a condition works:
```
actor Thing():
    def foo():
        return 1

actor main(env):
    var thingies = [Thing()]

    for thing in thingies:
        bar = "hello"
        break

    env.exit(0)
```
```
kll@ThinkYoga:~/terastream/acton/test$ ../actonc --root main for-break.act
kll@ThinkYoga:~/terastream/acton/test$ echo $?
0
kll@ThinkYoga:~/terastream/acton/test$
```

--- a/compiler/Acton/CPS.hs
+++ b/compiler/Acton/CPS.hs
@@ -380,6 +380,8 @@ instance NeedCont Branch where
 
 instance NeedCont Stmt where
     needCont env (Return _ (Just e))    = inCont env
+    needCont env (Continue _)           = inCont env
+    needCont env (Break _)              = inCont env
     needCont env (Expr _ e)             = contCall env e
     needCont env (Assign _ _ e)         = contCall env e
     needCont env (MutAssign _ _ e)      = contCall env e

--- a/test/regression/Makefile
+++ b/test/regression/Makefile
@@ -12,7 +12,6 @@ TESTS=$(TEST_SOURCES:.act=)
 # during compilation using actonc or at run time.
 FAILING_COMPILATION= \
 	11-int-float \
-	20-for-break \
 	rtail \
 	quotes-in-triquotes \
 	triquotes-across-lines
@@ -52,8 +51,5 @@ $(FAILING_RUNNING):
 	@echo "Now we really see 2 'pong' messages"
 	./$@
 	./$@ | grep -c pong | grep 2
-
-210-nontermination:
-	@echo "Non-termination test disabled (for obvious reasons...)"
 
 .PHONY: test test-examples $(TESTS)

